astropy/table/table.py

```diff
@@ -335,6 +362,88 @@ def __get__(self, instance, owner_cls):
         return info
 
 
+class HiddenColumns(MetaAttribute):
+    """Maintain tuple of table column names that are hidden from print output.
+
+    This is a descriptor that inherits from MetaAttribute so that the attribute
+    value is stored in the table meta['__attributes__'].
+    """
+    def __get__(self, instance, owner_cls):
+        # For getting from class not an instance
+        if instance is None:
+            return self
+
+        hidden_columns = instance.__dict__.get('hidden_columns')
+        if hidden_columns is None:
+            hidden_columns = deepcopy(self)
+            instance.__dict__['hidden_columns'] = hidden_columns
+        # We set _parent_ref on every call, since if one makes copies of
+        # instances, 'hidden_columns' will be copied as well, which will lose the
+        # reference.
+        hidden_columns._instance_ref = weakref.ref(instance)
+        return hidden_columns
+
+    def __set__(self, instance, names):
+        if isinstance(names, str):
+            names = [names]
+        if names is None:
+            # Remove hidden_columns from the meta['__attributes__'] dict.
+            # Subsequent access will just return None.
+            del instance.hidden_columns
+        else:
+            return super().__set__(instance, tuple(names))
+
+    def __call__(self):
+        instance = self._instance_ref()
+        return super().__get__(instance, instance.__class__)
```

Comment:
> Would it make sense to add or () at the end so that one always gets a tuple?

This would break the nice implementation of `set_temp()` because it would morph an original value of `None` to a final value of `()`.